README.md

```diff
@@ -1,2 +1,555 @@
+To override this behavior, the `prefix` parameter of either `Flag` or `Opt` annotations should be used.
+
+Consider the following example:
+```py
+# test.py
+
+from {pyname} import cli, Flag
+
+@cli
+def entry(some: Flag(prefix='+')):
+    print(some)
+
+if __name__ == '__main__':
+    cli()
+```
+
+Then the command line interface could be invoked as follows:
+```
+$ python test.py ++some
+True
+```
+
+### Types
+
+By default, the value that comes from the CLI, if it's an `Opt` or an `Arg`, is of `str` type.
+
+Consider the following example:
+```py
+# test.py
+
+from {pyname} import cli, Arg
+
+@cli
+def entry(some: Arg):
+    print(type(some))
+
+if __name__ == '__main__':
+    cli()
+```
+
+If an integer value is passed to the command line, the following output is produced:
+```
+$ python test.py 1
+<class 'str'>
+```
+
+To enforce the value to be an `int`, the following syntax should be used:
+```py
+# test.py
+
+from {pyname} import cli, Arg
+
+@cli
+def entry(some: Arg[int]):
+    print(type(some))
+
+if __name__ == '__main__':
+    cli()
+```
+
+This makes `some` to be an integer:
+```
+$ python test.py 1
+<class 'int'>
+```
+
+But sometimes, a custom type should be parsed from a command line.
+This could be done by defining a parser for one using the `parse` decorator:
+```py
+# test.py
+
+from {pyname} import cli, Arg
+
+class Custom:
+    ...
+
+@cli
+def entry(some: Arg[Custom]):
+    ...
+
+@cli.parse
+def custom(x: str) -> Custom:
+    ...
+
+if __name__ == '__main__':
+    cli()
+```
+
+Note: it's possible to manually add a parser for a type using the `add_parser` function.
+It works in the same way as the `parse` decorator:
+```py
+# test.py
+
+from {pyname} import cli, Arg
+
+class Custom:
+    def parse(x: str) -> 'Custom':
+        ...
+
+@cli
+def entry(some: Arg[Custom]):
+    ...
+
+cli.add_parser(Custom.parse)
+
+if __name__ == '__main__':
+    cli()
+```
+
+### Commands
+
+Complex command line interfaces like `git` have several subcommands: `git status`, `git pull`, `git push`, etc.
+These subcommands act as separate CLIs and, thus, are separate functions in {Name}.
+
+Consider the following example as a fake `git` CLI:
+```py
+# git.py
+
+from {pyname} import cli
+
+@cli
+def entry():
+    ...
+
+@entry.command
+def pull():
+    if rebase:
+        print('pulling with rebase')
+    else:
+        print('pulling')
+
+@entry.command
+def push():
+    print('pushing')
+
+if __name__ == '__main__':
+    cli()
+```
+
+This CLI could be invoked as follows:
+```
+$ python git.py pull
+pulling
+$ python git.py pull --rebase
+pulling with rebase
+$ python git.py push
+pushing
+```
+
+Note: it's possible to have deeper hierarchies of subcommands.
+For example, the `dotnet` CLI tool allows [`dotnet tool install {x}`](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-tool-install) syntax.
+
+This could be implemented in {Name} in a natural way:
+```py
+# dotnet.py
+
+from {pyname} import cli
+
+@cli
+def entry():
```

Comment:
Maybe `dotnet` instead of `entry`? :)

Reply:
I wanted to be consistent across all examples. Do you think this is not that important here?

Reply:
Oh, I now see that when we use `help`, the name of the function is printed, so I've also changed the `entry` to `dotnet` (as well as `entry` to `git`).